docs: Add Open Term Loan Manager section (SC-12230) (#39)
maple-labs/maple-docs: technical-resources/loan-managers/open-term-loan-manager/open-term-loan-manager.md

# Impairment / Remove Impairment 
- Both the PoolDelegate and Governor can call `impairLoan()` which calls the Loan's `impair()` and updates the LoanManager accounting by reducing the issuanceRate, updates the `unrealizedLosses` with the principal and accrued interest and stores the impairment information in the `Impairment` struct for future use. 
- Both the PoolDelegate and Governor can call `removeLoanImpairment()` but if the Loan was Impaired by the Governor only the Governor can call `removeLoanImpairment()`, this is to safeguard against any malicious use. In the case `removeLoanImpairment()` is called the LoanManager accounting is updated to include the issuanceRate for the Loan payment and add back any Interest that was expected to be accrued during this period between the Impairment date and the current timestamp. Finally `removeImpairment()` is called on the Loan. 

# Default
- The PoolManager can call `triggerDefault()` on the LoanManager. This is due to the PoolManager also needing to handle PoolCover so the decision was made to keep the entry point to `triggerDefault()` the PoolManager. 

> - Note the interface for `triggerDefault(address loan_, address liquidatorFactory_)` takes in a liquidatorFactory address this is ignored for the execution in the Open-Term LoanManager and is in the interface for compatibility of the PoolManager with both Fixed/Open-Term LoanManagers. 

Reviewer: ```suggestion
- Note the interface for `triggerDefault(address loan_, address liquidatorFactory_)` takes in a `liquidatorFactory` address, which is ignored for the execution in the Open-Term LoanManager (since Open-Term Loans have no collateral) and is in the interface for compatibility of the PoolManager with both Fixed/Open-Term LoanManagers. 
```

> - As `triggerDefault()` could have many states a Loan is in prior to defaulting such as being late or Impaired the decision was made to ensure the Loan state is idempotent and is always Impaired first as part of the `triggerDefault()` flow. 

Reviewer: ```suggestion
- The natural progression of all defaults is for the Loan to be impaired first, such that the calculation during the default process to occur from the same station Loan accounting state. Thus, `triggerDefault()` performs steps to account for the Loan's impairment first, if it has already not been impaired, as part of the `triggerDefault()` flow. 
```

- Global accounting is updated to update the `principalOut`, `accountedInterest` and `unrealizedLosses`.
- Any recovered fees from the Loan are distributed to the Pool and Treasury with any `remainingLosses` passed back to the `PoolManager.triggerDefault()` to see if anything further can be recovered via the PoolCover.    

# Fees

> - Service fees are passed into the `claim()` function and are paid out to the platform and delegate respectively. It is worth noting that if the poolDelegate does not have enough cover provided the delegateServiceFees are added to the platformFees and sent to the treasury (to be later determined what to do with).

Reviewer: ```suggestion
- Service fees are passed into the `claim()` function and are paid out to the platform and delegate respectively. It is worth noting that if the PoolDelegate does not have enough cover provided, the `delegateServiceFees` are added to the `platformFees` and sent to the Treasury.
```

> - Management fees are calculated as part of the `claim()` function flow using the stored managementFeeRate in the `Payment` struct. Similar to service fees, if the poolDelegate does not provide enough cover the delegateManagementFee is instead not deducted and the pool gets that portion of fees.  

Reviewer: ```suggestion
- Management fees are calculated as part of the `claim()` function flow using the stored `managementFeeRate` in the `Payment` struct. Similar to service fees, if the PoolDelegate does not provide enough cover, the `delegateManagementFee` is instead not deducted and the pool gets that portion of fees.  
```

# Accounting  

> The Open Term Loan Manager handles accounting for Open Term Loans by being aware of the incoming payment expected. When a Loan if Funded or a Payment is made the payment information is stored in a payment struct containing the `platformManagementFeeRate`, `delegateManagementFeeRate`, `startDate` and `issuanceRate` with the latter 2 being Loan specific vs the global variables. The global `issuanceRate` is then adjusted and interest accrued prior gets added to the `accountedInterest` thus updating the accounting state globally. 

Reviewer: ```suggestion
The Open Term Loan Manager handles accounting for Open Term Loans by being aware of the incoming payment expected. When a Loan if Funded or a Payment is made the payment information is stored in a payment struct containing the `platformManagementFeeRate`, `delegateManagementFeeRate`, `startDate` and `issuanceRate`. The fee rates are snapshots of the global variables at the time, whereas the latter 2 are Loan specific. The global `issuanceRate` is then adjusted and interest accrued prior gets added to the `accountedInterest`, thus updating the accounting state globally. 
```

> Note the key distinction between the Open Term Loan Manager and Fixed Term Loan Manager is that in the case of Open Term interest keeps accruing as there is no notion of a `domainEnd`, the assumption is that the Pool Delegate will act to Impair or Default a Open Term Loan in order to make the necessary accounting updates to adjust the issuance rate and stop accruing interest on a Loan where a payment is no longer expected.

Reviewer: ```suggestion
Note the key distinction between the Open Term Loan Manager and Fixed Term Loan Manager is that in the case of Open Term interest keeps accruing as there is no notion of a `domainEnd`. The assumption is that the Pool Delegate will act to Impair or Default a Open Term Loan in order to make the necessary accounting updates to adjust the issuance rate and stop accruing interest on a Loan where a payment is no longer expected.
```

# Accounting Examples

> Below please find examples of how the accounting state changes based on different scenarios. The scenarios assume the Pool only has Open-Term Loans via the Open-Term LoanManager.

Reviewer: ```suggestion
See below for examples of how the accounting state changes based on different scenarios. The scenarios assume the Pool only has Open-Term Loans via the Open-Term LoanManager.
```
